### PR TITLE
new branch

### PR DESCRIPTION
**Pull request by HaoYang0000**
PR for https://jira.verticacorp.com/jira/browse/VER-95581

### CODE REVIEW

File: .github/workflows/daily-work.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+on:
+  workflow_run:
+    workflows: ['e2e tests']
+    types: [completed]
+
+jobs:
+  build-operator:
+    runs-on: ubuntu-latest 
+    if: ${{ github.event.workflow_run.conclusion == 'success' && contains('server daily version', github.event.workflow_run.display_title) }}
```
Comment (cchen-vertica) [inline .github/workflows/daily-work.yaml:11]: Check if the runner is "verticareleng" too

File: .github/workflows/daily-work.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: Daily work
+
+on:
+  workflow_run:
+    workflows: ['e2e tests']
+    types: [completed]
+
+jobs:
+  build-operator:
```
Comment (cchen-vertica) [inline .github/workflows/daily-work.yaml:9]: Rename it to "upload-operator-to-private-repo"

File: .github/workflows/daily-work.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Upload new image
+      run: |
+        echo "Pushing images with tag: $VERSION_SHA"
+        # FIXME: uncomment the following
+        # docker buildx imagetools create --tag opentext/verticadb-operator:$VERSION ghcr.io/vertica/verticadb-operator:$VERSION_SHA
```
Comment (cchen-vertica) [inline .github/workflows/daily-work.yaml:47]: Use the repo "opentext/verticadb-operator-private" to replace "opentext/verticadb-operator"

File: .github/workflows/daily-work.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  workflow_run:
+    workflows: ['e2e tests']
+    types: [completed]
+
+jobs:
+  build-operator:
+    runs-on: ubuntu-latest 
+    if: ${{ github.event.workflow_run.conclusion == 'success' && contains('server daily version', github.event.workflow_run.display_title) }}  
+    steps:
+    - uses: actions/checkout@v4
```
Comment (cchen-vertica) [inline .github/workflows/daily-work.yaml:13]: This doesn't need to be in a new workflow. Just add one step in e2e-tests, then get the operator version from "build.outputs.operator-image". We don't need to download the repo and infer the versions again.

File: .github/workflows/daily-work.yaml (hunk 3 above)
Comment (HaoYang0000) [reply .github/workflows/daily-work.yaml]: Updated in https://github.com/vertica/vertica-kubernetes/pull/929/commits/cc05ac3845e2ed5b6da9410247a8d3b3751254e4

File: .github/workflows/daily-work.yaml (hunk 2 above)
Comment (HaoYang0000) [reply .github/workflows/daily-work.yaml]: Updated in https://github.com/vertica/vertica-kubernetes/pull/929/commits/c49e341eb3a46af8b91b01edcc634c0f284322d1

File: .github/workflows/daily-work.yaml (hunk 1 above)
Comment (HaoYang0000) [reply .github/workflows/daily-work.yaml]: Updated in https://github.com/vertica/vertica-kubernetes/pull/929/commits/0533fe0dffa257580ad8034125d2e04e2524fa06

File: .github/workflows/daily-work.yaml (hunk 4 above)
Comment (HaoYang0000) [reply .github/workflows/daily-work.yaml:13]: I thought we want to build a new daily work action to control the upload of the image. It is actually much easier to just add one new job in the e2e.yml. If that is what we want I will do that, please let me know

Comment (HaoYang0000) [reply .github/workflows/daily-work.yaml]: Moved the task to e2e.yml in https://github.com/vertica/vertica-kubernetes/pull/929/commits/61989734fa878a5b2719d6dcf2f048ae4a10a3be

File: .github/workflows/e2e.yml [hunk 5]
```diff
@@ -638,6 +638,57 @@ jobs:
...
+    ]
+    runs-on: ubuntu-latest 
+    if: ${{ always() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') && contains('server daily version', github.event.workflow_run.display_title) && contains('verticareleng', github.actor) }}  
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        path: vertica-kubernetes
+        fetch-depth: 0
+
+    - name: Save current version to env VERSION
```
Comment (cchen-vertica) [inline .github/workflows/e2e.yml:669]: We don't need to infer the version again(remove this step). You can either get the version from "build.outputs.operator-image"(I perfer this way) or "github.sha".

File: .github/workflows/e2e.yml [hunk 6]
```diff
@@ -638,6 +638,57 @@ jobs:
           vertica-deployment-method: vclusterops
           communal-storage-type: s3
   
+  upload-operator-to-private-repo:
+    needs: [
+      build,
+      e2e-leg-1-vcluster,
```
[the remark this answers is not in the document]
Comment (HaoYang0000) [reply .github/workflows/e2e.yml:644]: We have the following jobs in the e2e.yml: e2e-leg-1-admintools, e2e-leg-1-admintools-previous-release, e2e-leg-2-admintools, e2e-leg-2-admintools-previous-release, e2e-leg-3-admintools, e2e-leg-4-admintools, e2e-leg-5-admintools, e2e-server-upgrade-admintools, e2e-udx-admintools. 
I checked the last daily e2e run, other than the previous-release ones, do we want to include all of them? If confirmed I will add it

File: .github/workflows/e2e.yml (hunk 5 above)
Comment (HaoYang0000) [reply .github/workflows/e2e.yml:669]: The value of "build.outputs.operator-image": is something like: ghcr.io/vertica/verticadb-operator:93d02b6d9356416a54317e291a0638163b84b181, which contains both repo and tag, not the version value. And what we are looking for here is the operator version, something like: VERSION=24.4.0-0. And we need both version and version-sha

File: .github/workflows/e2e.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (HaoYang0000) [reply .github/workflows/e2e.yml:644]: Per discussion, steps added in https://github.com/vertica/vertica-kubernetes/pull/929/commits/f8e7628400a3e0c2722ba7cec32de12a35b6a48e

File: .github/workflows/e2e.yml (hunk 5 above)
Comment (HaoYang0000) [reply .github/workflows/e2e.yml]: Per discussion, used the operator-image output, changed updated in https://github.com/vertica/vertica-kubernetes/pull/929/commits/53074ab1b34e498ad3e6c0e6b2ac2f2e4d8c319a

**Review by cchen-vertica — APPROVED** [review]

**Pull request merged**